gui/default/syncthing/core/syncthingController.js

```diff
@@ -912,6 +923,17 @@ angular.module('syncthing.core')
             return '';
         };
 
+        $scope.findDeviceFromShort = function (shortID) {
+            var matches = $scope.devices.filter(function (n) {
+                return n.deviceID.substr(0, 7) === shortID;
+            });
+            if (matches.length !== 1) {
+                n.deviceID.name = "Unknown";
+                return n;
+            }
+            return matches[0];
+        };
+
```

Comment:
This is a bad idea, you are modifying attributes on a random object that is persisted, so this will affect other things.

I'd replace:
`{{deviceName(findDeviceFromShort(changeEvent.data.modifiedBy))}}`

With:
```
{{deviceNameFromShort(changeEvent.data.modifiedBy))}}
```

and then just do:

```
for (i= 0; i< a.length; i++) {
    if ($scope.devices[i].deviceID.substr(0, 7) == shortID) {
        return $scope.devices[i].name;
    }
}
return 'Unknown';
```